library/billing/src/Claim.php

```diff
@@ -53,6 +39,26 @@ class Claim
     var $copay;             // total of copays from the ar_activity table
     var $facilityService;
 
+    // This enforces the X12 Basic Character Set. Page A2.
+    private function x12clean($str)
+    {
+        return preg_replace('/[^A-Z0-9!"\\&\'()+,\\-.\\/;?= ]/', '', strtoupper($str));
+    }
+
+// Make sure dates have no formatting and zero filled becomes blank
+// Handles date time stamp formats as well
+
+    private function cleanDate($date_field)
```

Comment:
phpstorm pretty much does everything one can imagine. Here's a nice video that @robertdown  sent me which has been helpful: https://www.youtube.com/watch?v=TejBj_N-3rI